test/unittest/blas1/blas1_dot_test.cpp

```diff
@@ -26,7 +26,8 @@
 #include "blas_test.hpp"
 
 template <typename scalar_t>
-using combination_t = std::tuple<std::string, api_type, int, int, int>;
+using combination_t =
+    std::tuple<std::string, api_type, int, int, int, scalar_t>;
```

Comment:
Could you use `index_t` instead of `int` here? I think that keep it generic would help us to enable tests for different index type in the future, if necessary.

Reply:
hmm fair enough but we usually don't template the test parameter combination_t with the `index_t`, I think it's better to have it in a separate PR in case we want to as it has to be reflected on all other operators tests.